core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAO.scala

```diff
@@ -0,0 +1,76 @@
+package org.broadinstitute.dsde.rawls.billing
+
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.config.{AzureConfig, MultiCloudWorkspaceConfig}
+import org.broadinstitute.dsde.rawls.dataaccess.SamDAO
+import org.broadinstitute.dsde.rawls.model.{CreationStatuses, RawlsBillingProject, RawlsBillingProjectName, RawlsBillingProjectResponse, SamResourceAction, SamResourceTypeNames, SamUserResource, UserInfo, AzureManagedAppCoordinates}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+ * Facade over the billing profile manager service. This service will eventually by the source of truth for
+ * billing profiles in the Terra system. For now, we are using this to layer in "external" billing profiles
+ * for the purposes of testing Azure workspaces.
+ */
+class BillingProfileManagerDAO(samDAO: SamDAO, config: MultiCloudWorkspaceConfig) extends LazyLogging {
```

Comment:
Will this work with terra ui billing project sharing?

Reply:
Answer: yes, in my testing as long as the sam resources are properly setup we can add owners + users.